build-tools/automation/azure-pipelines.yml

```diff
@@ -0,0 +1,215 @@
+# Xamarin.Android Pipeline
+
+trigger:
+  - master
+  - d16-?
+
+# Global variables
+variables:
+  BundleArtifactName: bundle
+  AutoProvisionArgs: /p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False
+
+stages:
+- stage: prepare
+  displayName: Prepare
+  variables:
+    Configuration: Release
+    MSBuildAbiArgs: >-
+      /p:AndroidSupportedTargetJitAbis=armeabi-v7a:arm64-v8a:x86:x86_64
+      /p:AndroidSupportedHostJitAbis=Darwin:mxe-Win32:mxe-Win64
+      /p:AndroidSupportedTargetAotAbis=armeabi-v7a:arm64:x86:x86_64:win-armeabi-v7a:win-arm64:win-x86:win-x86_64
+  jobs:
+  - job: create_bundle
+    displayName: Create Bundle
+    pool: Hosted Mac Internal
+    timeoutInMinutes: 60
+    cancelTimeoutInMinutes: 5
+    steps:
+    - script: make prepare-deps V=1 CONFIGURATION=$(Configuration) MSBUILD_ARGS="$(AutoProvisionArgs) $(MSBuildAbiArgs)"
+      displayName: make prepare-deps
+
+    - task: MSBuild@1
+      displayName: create bundle
+      inputs:
+        solution: build-tools/create-bundle/create-bundle.csproj
+        configuration: $(Configuration)
+        msbuildArguments: /bl:bin/Build$(Configuration)/msbuild-createbundle.binlog $(MSBuildAbiArgs)
+
+    - task: CopyFiles@2
+      displayName: copy bundle
+      inputs:
+        sourceFolder: bin/$(Configuration)
+        contents: bundle-*.zip
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: PublishPipelineArtifact@0
```

Comment:
Probably related: https://github.com/xamarin/xamarin-android/issues/3095

Currently, we always upload the bundle, even if it "hasn't changed."  We should reconsider this (hence the above issue).

Whether we should reconsider this *now* (with this PR) or *later*, I'm less certain about.

Can YAML have conditionals?

Reply:
The only alternative to this artifact upload approach that I can think of at the moment would be to partially re-implement the bundle upload logic in our groovy scripts that uses a separate azure blob storage container.

Rather than doing that, I think it would be worth investing in your previous suggestion of "NuGetizing" libzip, which would allow us to delete this step (and our bundle creation/download logic) entirely and only download/cache relevant mono archive files.

Reply:
We'll look to make this improvement in the future.